### PR TITLE
cql3: Specify arguments types in UDA creation errors

### PR DESCRIPTION
**Pull request by Jadw1**
Display not only function name but also expected arguments if `state_function` or `final_function` was not found.

Fixes: #12088

### CODE REVIEW

**Comment by nyh** [comment]
@Jadw1 I don't know why the CI run was "aborted", but it indicates several UDA-related test failures so may be related to your patch. Please take a look.

**Comment by nyh** [comment]
I checked, and some unit tests check the old message, e.g., `"State function not found"` not expecting the new message which can be `"State function.* not found"` (with something between the word "function" and "not found"). Please fix the tests in your patch too, and everything will be good (that's the beauty of having the tests in the same repository).

**Comment by avikivity** [comment]
Please use Fixes, not Closes, to avoid confusing our issue tracking machinery,

**Comment by Jadw1** [comment]
> Please use Fixes, not Closes, to avoid confusing our issue tracking machinery,

Changed

**Comment by avikivity** [comment]
The commit itself doesn't mention the issue at all.

File: cql3/statements/create_aggregate_statement.cc [hunk 1]
```diff
@@ -36,10 +36,19 @@ seastar::future<shared_ptr<db::functions::function>> create_aggregate_statement:
     acc_types.insert(acc_types.end(), _arg_types.begin(), _arg_types.end());
     auto state_func = dynamic_pointer_cast<functions::scalar_function>(functions::functions::find(functions::function_name{_name.keyspace, _sfunc}, acc_types));
     if (!state_func) {
-        throw exceptions::invalid_request_exception(format("State function not found: {}", _sfunc));
+        sstring args;
+        for (int i = 0; i < acc_types.size(); i++) {
```
Comment (nyh) [inline cql3/statements/create_aggregate_statement.cc:40]: Please consider using fmt::join instead of this loop. I think it makes the code nicer and shorter. E.g., you should be able to just print `fmt::join(acc_types, ", ")` and it will just do what you wanted.

File: test/cql-pytest/test_uda.py [hunk 2]
```diff
@@ -107,13 +107,13 @@ def test_incorrect_state_func(scylla_only, cql, test_keyspace):
     div_body = "(state tuple<bigint, bigint>) CALLED ON NULL INPUT RETURNS bigint LANGUAGE lua AS 'return state[1]//state[2]'"
     with new_function(cql, test_keyspace, avg_partial_body) as avg_partial, new_function(cql, test_keyspace, div_body) as div_fun:
         custom_avg_body = f"(bigint) SFUNC {avg_partial} STYPE tuple<bigint, bigint> FINALFUNC {div_fun} INITCOND (0,0)"
-        with pytest.raises(InvalidRequest, match="State function not found"):
+        with pytest.raises(InvalidRequest, match=r"State function (.*) not found"):
```
Comment (nyh) [inline test/cql-pytest/test_uda.py:110]: Nitpick: you don't need the parentheses in the regex - they don't do anything.
Also, why did you add the "r"? What is it supposed to achieve here?

File: test/cql-pytest/test_uda.py [hunk 3]
```diff
@@ -107,13 +107,13 @@ def test_incorrect_state_func(scylla_only, cql, test_keyspace):
...
             cql.execute(f"CREATE AGGREGATE {test_keyspace}.{unique_name()} {custom_avg_body}")
     avg2_partial_body = "(state tuple<bigint, bigint>) CALLED ON NULL INPUT RETURNS tuple<bigint, bigint> LANGUAGE lua AS 'return {state[1] + 42, state[2] + 1}'"
     div_body = "(state tuple<bigint, bigint>) CALLED ON NULL INPUT RETURNS bigint LANGUAGE lua AS 'return state[1]//state[2]'"
     with new_function(cql, test_keyspace, avg2_partial_body) as avg2_partial, new_function(cql, test_keyspace, div_body) as div_fun:
         custom_avg_body = f"(bigint) SFUNC {avg2_partial} STYPE tuple<bigint, bigint> FINALFUNC {div_fun} INITCOND (0,0)"
-        with pytest.raises(InvalidRequest, match="State function not found"):
+        with pytest.raises(InvalidRequest, match=r"State function (.*) not found"):
```
Comment (nyh) [inline test/cql-pytest/test_uda.py:116]: Again, these parentheses  don't do anything.

**Review by nyh — CHANGES_REQUESTED** [review]
Looks mostly good, I left a few minor comments, but please also modify the commit message itself (not the cover letter, which is ignored in a single-patch PR) to include the Fixes line. We need this for the issue to be closed automatically (and for backporting decisions).

File: test/cql-pytest/test_uda.py (hunk 2 above)
Comment (Jadw1) [reply test/cql-pytest/test_uda.py]: I've added the prefix by mistake, removed.

File: test/cql-pytest/test_uda.py (hunk 3 above)
Comment (Jadw1) [reply test/cql-pytest/test_uda.py]: removed

**Comment by Jadw1** [comment]
Did the changes and added "Fixes" to commit message

**Review by nyh — APPROVED** [review]
